### DISCUSSION TITLE
updating table from json state object

### DISCUSSION THREAD

**Discussion by pchanley** [Budibase/budibase#15060] [Help]
I have created a json object from 3 tables combined in a specific way.  I want to create a table from the json.  ideally would update in javascript.  any help would be greatly appreciated

**Comment by mikesealey** [comment]
Hey @pchanley 

When you add a table to your screen you can select the Data Source - Typically these would be tables from your Data Section, Views (or View Calculations (v3.0!)) or queries. Near the bottom of the list you can select JSON/CSV. You may find that the table component automatically selected one of the available options on your behalf, but you can simply change it in the settings. 

![image](https://github.com/user-attachments/assets/c0d9734d-a121-4a6b-8f0d-1e0786ea9592)

When you switch to JSON/CSV, you will see a little cog/gear/sprocket icon next to the setting. Clicking this will open the bindings drawer, where you can use the bindings to bring in your variables, write your JSON or CSV, or use a mixture of the two approaches to achieve your goal.

You can use the text input of the bindings drawer to simply write comma-separated values 
![image](https://github.com/user-attachments/assets/4e5f9822-ed5d-453a-a6f0-d712d87d8a8f)

Alternatively you can use the JavaScript input of the bindings drawer to return an array of objects, as such:

![image](https://github.com/user-attachments/assets/cff9b2cd-03ac-4674-87de-f31bebdc7a9f)

```
return [
  {name: "Mike", fav_colour: "Blue"},
  {name: "Conor", fav_colour: "Green"},
  {name: "Andy", fav_colour: "Red"},
 ]
```

If you're using the Internal BudibaseDB you can simply use a View to display related tables though, you don't need the above extra steps.

Go to the Data section, and view one of the tables in question.
Create a Table View, and name it.
![image](https://github.com/user-attachments/assets/8844f0e4-5bda-47be-9e15-0ad57032f007)
![image](https://github.com/user-attachments/assets/a232cf95-aab1-4a2e-b31d-eda105b2be0b)

Then click the Columns button, and where you have a relationship column you will be able to look into the other table. By default you'll see that all columns in the related table are hidden. 
![image](https://github.com/user-attachments/assets/fea63384-652b-4a7e-9bee-5c13b1013da3)

**Comment by pchanley** [comment]
Mike,

great info, much appreciated. 

 As i am creating the data for the table on the fly based a combination of data from 3 different files. i ampopulating the table on the fly.  and that works fine.  However, I would need to apply column attributes to the table.  is that possible.  Or if I define a table in the Buidbase and set up the appropriate attributes there, How would I update that table from my state array?

Thanks

Peter

**Comment by pchanley** [comment]
I am thinking that perhaps data table is not the right component for what I am trying to do.  as I have to add the ability to update entries.  and various buttons to update groups.  So perhaps you would have another suggestions.  perhaps multriple fors or field groups?

thanks

**Comment by pchanley** [comment]
can you update a budibase table from a state array.
